### PR TITLE
[ Milestone 10 ] Authorization

### PR DESCRIPTION
**Pull request by CakeInTech**
- Installed CanCanCan in your project.
- Added a role column to the users table. Remember to use a migration for this.

- [x] A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- [x] For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- [x] A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- [x] For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    if user.is?(:admin)
+      can :manage, :all
+    else
+      can :read, :all
+      can :destroy, Post, author: user
+      can :destroy, Comment, author: user
```
Comment (thecodechaser) [inline app/models/ability.rb]: - [x] Please make sure to use `:manage` instead `:destroy` for posts and comments for their authors. `:manage` will allow the authors to perform all actions view, update and delete.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -20,4 +20,10 @@ def show
     @post = Post.find(request.parameters[:id])
     @user = User.find(params[:user_id])
   end
+
+  def destroy
+    @post = Post.find(request.parameters[:id])
+    @post.destroy
+    redirect_to all_post_path(params[:user_id])
+  end
 end
```
Comment (thecodechaser) [inline app/controllers/posts_controller.rb:30]: - [x] Please make sure to use `CanCanCan` for Post authorization in Post Controller. You can consider the below code.

```ruby
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @MohamedCK, and @eerapheal👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Highly organized code. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (CakeInTech) [reply app/models/ability.rb]: - Thank you for the suggestion, I have implemented it. 😄

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello @MohamedCK   , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: aaa
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R
- [x] Proffessional README









 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**